File: app/src/main/java/com/feliperamoscarvalho/appconsultagithub/data/RetrofitClient.java
package com.feliperamoscarvalho.appconsultagithub.data;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Class that configures the retrofit for use
 */
public class RetrofitClient {

    public static final String BASE_URL = "https://api.github.com";

    private static Retrofit retrofit = null;

    private static Gson gson = new GsonBuilder().create();

    /**
     * Starts the retrofit object
     */
    public static Retrofit getClient() {
        if(retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();
        }

        return retrofit;
    }
}

File: app/src/main/java/com/concretesolutions/diegosouza/concretesolutions/Api/RetrofitClient.kt
package com.concretesolutions.diegosouza.concretesolutions.Api

import com.concretesolutions.diegosouza.concretesolutions.Service.ListaService
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitClient {

    private val retrofit = Retrofit.Builder()
        .baseUrl("https://api.github.com/search/")
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    fun listaService() = retrofit.create(ListaService::class.java)
}